packages/components/src/modifiers/hds-anchored-position.ts

```diff
@@ -0,0 +1,274 @@
+  // size({
+  //   apply({ rects, elements }) {
+  //     Object.assign(elements.floating.style, {
+  //       width: `${rects.reference.width}px`,
+  //     });
+  //   },
+  // });
+  // size({
+  //   apply: ({ availableWidth, availableHeight, middlewareData }) => {
+  //     middlewareData.size = { availableWidth, availableHeight };
+  //   },
+  // }),
+
+  middleware.push(...middlewareExtra);
+
+  return {
+    placement,
+    strategy,
+    middleware,
+  };
+};
+
+interface HdsAnchoredPositionSignature {
+  Element: HTMLElement;
+  Args: {
+    Positional: [HTMLElement | SVGElement];
+    Named: FloatingUIOptions & { arrowSelector?: string };
+  };
+}
+
+// Notice: we use a function-based modifier here instead of a class-based one
+// because it's quite simple in its logic, and doesn't require injecting services
+// see: https://github.com/ember-modifier/ember-modifier#function-based-modifiers
+
+export default modifier<HdsAnchoredPositionSignature>(
+  (element, positional, named = {}) => {
+    // the element that "floats" next to the "anchor" (whose position is calculated in relation to the anchor)
+    // notice: this is the element the Ember modifier is attached to
+    const floatingElement = element;
+
+    // the element that acts as an "anchor" for the "floating" element
+    // it can be a DOM (string) selector or a DOM element
+    // notice: it's expressed as "positional" argument (array of arguments) for the modifier
+    const _anchorTarget = positional[0];
+    const anchorElement =
+      typeof _anchorTarget === 'string'
+        ? document.querySelector(_anchorTarget)
+        : _anchorTarget;
+
+    assert(
+      '`hds-anchored-position` modifier - the provided "anchoring" element is not defined correctly',
+      anchorElement instanceof HTMLElement ||
+        anchorElement instanceof SVGElement
+    );
+
+    // the "arrow" element (optional) associated with the "floating" element
+    // it can be a DOM selector (string) or a DOM element
+    // notice: it's declared inside the "named" argument (object) for the modifier
+    // but we need to extract it also here so it can be used to assign inline styles to it
+    let arrowElement: HTMLElement | SVGElement | undefined;
+
+    if (named.arrowElement) {
+      assert(
+        '`hds-anchored-position` modifier - the `element` provided for the "arrow" element is not a valid DOM node',
+        named.arrowElement instanceof HTMLElement ||
+          named.arrowElement instanceof SVGElement
+      );
+
+      arrowElement = named.arrowElement;
+    } else if (named.arrowSelector) {
+      assert(
+        '`hds-anchored-position` modifier - the `selector` provided for the "arrow" element must be a string',
+        typeof named.arrowSelector === 'string'
+      );
+
+      const selectedArrowElement = document.querySelector(named.arrowSelector);
+      if (selectedArrowElement) {
+        arrowElement = selectedArrowElement as HTMLElement;
```

Comment:
We can avoid the type assertion here by stating the expected type at declaration
```suggestion
      const selectedArrowElement: HTMLElement | null = document.querySelector(
        named.arrowSelector
      );
      if (selectedArrowElement) {
        arrowElement = selectedArrowElement;
```

Reply:
or we can use a type guard in the conditional statement, however makes more sense to you
```js
if (selectedArrowElement instanceof HTMLElement) {
```

Reply:
I like the second one, thank you for the suggestion!